Platform/DeprecationWarner.swift

```diff
@@ -10,13 +10,39 @@ import Foundation
 
 #if DEBUG
     class DeprecationWarner {
-        private static var warned = [String]()
-
-        static func warnIfNeeded(_ message: String) {
-            guard !warned.contains(message) else { return }
-
-            warned.append(message)
-            print("ℹ️ [DEPRECATED] \(message)")
+        private static var warned = [Kind]()
+        private static var _lock = NSRecursiveLock()
+        
+        static func warnIfNeeded(_ kind: Kind) {
+            _lock.lock(); defer { _lock.unlock() }
+            guard !warned.contains(kind) else { return }
+            
+            warned.append(kind)
+            print("ℹ️ [DEPRECATED] \(kind.message)")
+        }
+    }
+    
+    extension DeprecationWarner {
+        enum Kind {
+            case variable
+            case globalTestFunction(String)
+            
+            var message: String {
+                switch self {
+                case .variable: return "`Variable` is planned for future deprecation. Please consider `RxCocoa.BehaviorRelay` as a replacement. Read more at: https://git.io/vNqvx"
```

Comment:
We can remove the specificity of `RxCocoa.BehaviorRelay` as just `BehaviorRelay` and then figure it out as we move towards the next major version